components/utils/DataProvider.js

```diff
@@ -13,34 +15,87 @@ class DataProvider extends Component {
       networks: ["Testnet One", "Testnet Two"],
       currentNetwork: null,
       updateNetwork: this.updateNetwork,
-      details: dummyData.details,
-      blocks: dummyData.blocks,
+      details: this.props.details,
+      blocks: this.props.blocks,
       transactions: dummyData.transactions
     };
   }
 
-  componentWillMount() {
+  static async getInitialProps(ctx) {
+    const [details, blocks] = await Promise.all([
+      DataProvider.getDetails(),
+      DataProvider.getBlocksInfo()
+    ]);
+    return {
+      details,
+      blocks
+    };
+  }
+
+  componentDidMount() {
```

Comment:
@frol Why make the call in `componentDidMount` instead of `componentWillMount`?
Does the backend call take that much time?

Reply:
`componentWillMount` is considered legacy and unsafe: https://reactjs.org/docs/react-component.html#the-component-lifecycle

Reply:
The reason I used `componentWillMount` is that I wanted the initial values populated before the render happens (rather than seeing the render happening, and then immediately rendering it again with the updated values).

I would have preferred doing the initialization in `constructor` if I could have.

But now I see you have some async calls happening. Then `componentDidMount` makes more sense.

Reply:
When you need something server-rendered, you should use `static async getInitialProps` (next.js built-in "extension"; just keep in mind that it works out of the box only on the page components and you have to manually call it on child components if you need it down the line). `componentWillMount` is not blocking.